### PR TITLE
[test-only] ApiTest. create tag

### PR DESCRIPTION
**Pull request by ScharfViktor**
- created api test  for adding a tag to a resource in the project space and personal space
- created method of getting tags
- created method of deleting tags (tests will be in the next PR)
- added checking the PROPFIND response for the presence of a tag on the resource
- ~~created bug #5384~~
- deleted code from core

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiAsyncUpload-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/18242/48/1) failed. Further test are cancelled...

**Comment by ScharfViktor** [comment]
It looks like that I fixed test, but how?
https://drone.owncloud.com/owncloud/ocis/18244/57/6
image

File: tests/acceptance/features/apiSpaces/tag.feature [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    And using spaces DAV path
+    And the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "use-tag" with the default quota using the GraphApi
+    And user "Alice" has created a folder "folderMain" in space "use-tag"
+    And user "Alice" has uploaded a file inside space "use-tag" with content "some content" to "folderMain/insideTheFolder.txt"
+
+
+  Scenario: Alice creates tags for resources in the project space
+    Given user "Alice" has shared a space "use-tag" to user "Brian" with role "viewer"
+    When user "Alice" creates the following tags a folder "folderMain" of the space "use-tag":
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/tag.feature:22]:
```suggestion
    When user "Alice" creates the following tags for folder "folderMain" of space "use-tag":
```

**Comment by ScharfViktor** [comment]
### deleted tests from `search.feature`
and closed bug https://github.com/owncloud/ocis/issues/3092
```
Scenario: search for entry by tags using REPORT method
    Given user "Alice" has created a "normal" tag with name "JustARegularTag1"
    And user "Alice" has created a "normal" tag with name "JustARegularTag2"
    And user "Alice" has added tag "JustARegularTag1" to folder "फनी näme"
    And user "Alice" has added tag "JustARegularTag1" to file "upload.txt"
    And user "Alice" has added tag "JustARegularTag2" to file "upload.txt"
    When user "Alice" searches for resources tagged with tag "JustARegularTag1" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Alice" should contain these entries:
      | फनी näme   |
      | upload.txt |
    When user "Alice" searches for resources tagged with tag "JustARegularTag2" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Alice" should contain these entries:
      | upload.txt |


  Scenario: share a tagged resource to another internal user and sharee searches for tag using REPORT method
    Given user "Brian" has been created with default attributes and without skeleton files
    And user "Alice" has created a "normal" tag with name "JustARegularTag1"
    And user "Alice" has created a "normal" tag with name "JustARegularTag2"
    And user "Alice" has added tag "JustARegularTag1" to folder "फनी näme"
    And user "Alice" has added tag "JustARegularTag1" to file "upload.txt"
    And user "Alice" has added tag "JustARegularTag2" to file "upload.txt"
    And user "Alice" has shared file "फनी näme" with user "Brian"
    And user "Alice" has shared file "upload.txt" with user "Brian"
    When user "Brian" searches for resources tagged with tag "JustARegularTag1" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Brian" should contain these entries:
      | फनी näme   |
      | upload.txt |
    When user "Brian" searches for resources tagged with tag "JustARegularTag2" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Brian" should contain these entries:
      | upload.txt |
    When user "Brian" searches for resources tagged with all of the following tags using the webDAV API
      | JustARegularTag1 |
      | JustARegularTag2 |
    Then the HTTP status code should be "207"
    And as user "Brian" the response should contain file "upload.txt"
    And as user "Brian" the response should not contain file "फनी näme"


  Scenario: search for entries across various folders by tags using REPORT method
    Given user "Alice" has created folder "/just-a-folder/inner-folder"
    And user "Alice" has uploaded file with content "inner file" to "/just-a-folder/inner-folder/upload.txt"
    And user "Alice" has created a "normal" tag with name "JustARegularTag1"
    And user "Alice" has created a "normal" tag with name "JustARegularTag2"
    And user "Alice" has added tag "JustARegularTag1" to folder "/just-a-folder/upload.txt"
    And user "Alice" has added tag "JustARegularTag1" to file "/फनी näme/upload.txt"
    And user "Alice" has added tag "JustARegularTag1" to file "/just-a-folder/inner-folder/upload.txt"
    And user "Alice" has added tag "JustARegularTag2" to file "/upload😀 😁/upload,1.txt"
    And user "Alice" has added tag "JustARegularTag2" to file "/upload.txt"
    When user "Alice" searches for resources tagged with tag "JustARegularTag1" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Alice" should contain these entries:
      | upload.txt |
      | upload.txt |
      | upload.txt |
    When user "Alice" searches for resources tagged with tag "JustARegularTag2" using the webDAV API
    Then the HTTP status code should be "207"
    And the search result by tags for user "Alice" should contain these entries:
      | upload,1.txt |
      | upload.txt   |
```

File: tests/acceptance/features/bootstrap/TagContext.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	 * @Then the following tag(s) should exist for user :user:
+	 *
+	 * @param string $user
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theFollowingTagsShouldExistForUser(string $user, TableNode $table):void {
+		$this->featureContext->setResponse(
```
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/acceptance/features/bootstrap/TagContext.php:130]: Thanks I split method

**Review by saw-jan — COMMENTED** [review]
Looks goof :+1:

File: tests/acceptance/features/apiSpaces/tag.feature [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    Then the HTTP status code should be "200"
+    When user "Alice" creates the following tags for file "file.txt" of space "Alice Hansen":
+      | fileTag                       |
+      | tag with symbol @^$#^%$@%!_+) |
+    Then the HTTP status code should be "200"
+    When user "Alice" sends PROPFIND request from the space "Alice Hansen" to the resource "folderMain" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the "PROPFIND" response to user "Alice" should contain a mountpoint "Alice Hansen" with these key and value pairs:
+      | key     | value            |
+      | oc:tags | my tag,important |
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/tag.feature:68]: what about combining those steps in to a single Then-step. e.g.
`@Then /^(?:file|folder|entry) "([^"]*)" of space "([^"]*)" should have the following tags for user "([^"]*)"$/`
then we would not need to repeat When-Then-And

File: tests/acceptance/features/bootstrap/TagContext.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			$user,
+			$this->featureContext->getPasswordForUser($user),
+			$resourceId,
+			$tagNameArray
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" has created the following tags a (folder|file)\s?"([^"]*)" of the space "([^"]*)":$/
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TagContext.php:104]:
```suggestion
	 * @Given /^user "([^"]*)" has created the following tags a (folder|file)\s?"([^"]*)" of the space "([^"]*)":$/
```

File: tests/acceptance/features/apiSpaces/tag.feature [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    When user "Alice" sends PROPFIND request from the space "Alice Hansen" to the resource "file.txt" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the "PROPFIND" response to user "Alice" should contain a mountpoint "Alice Hansen" with these key and value pairs:
+      | key     | value                                 |
+      | oc:tags | fileTag,tag with symbol @^$#^%$@%!_+) |
+    When user "Alice" lists all available tags via the GraphApi
+    Then the HTTP status code should be "200"
+    And the user should have following tags:
```
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpaces/tag.feature:76]: I splited this step after review https://github.com/owncloud/ocis/pull/5391#discussion_r1070869863

File: tests/acceptance/features/apiSpaces/tag.feature (hunk 3 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpaces/tag.feature:68]: I'm not sure. I wanted to make the steps more transparent and show in the test that I use propfind request

**Review by individual-it — APPROVED** [review]

**Pull request merged**